neurolib/models/multimodel/builder/hopf.py

```diff
@@ -0,0 +1,178 @@
+"""
+Hopf normal form model.
+
+References:
+    Kuznetsov, Y. A. (2013). Elements of applied bifurcation theory (Vol. 112).
+    Springer Science & Business Media.
+
+    Deco, G., Cabral, J., Woolrich, M. W., Stevner, A. B., Van Hartevelt, T. J.,
+    & Kringelbach, M. L. (2017). Single or multiple frequency generators in
+    on-going brain activity: A mechanistic whole-brain model of empirical MEG
+    data. Neuroimage, 152, 538-550.
+"""
+
+import numpy as np
+import symengine as se
+from jitcdde import input as system_input
+
+from ..builder.base.network import Network, Node
+from ..builder.base.neural_mass import NeuralMass
+
+DEFAULT_PARAMS = {
+    "a": 0.25,
+    "omega": 0.2,
+    "ext_input_x": 0.0,
+    "ext_input_y": 0.0,
+}
+
+
+class HopfMass(NeuralMass):
+    """
+    Hopf normal form (Landau-Stuart oscillator).
+    """
+
+    name = "Hopf normal form mass"
+    label = "HopfMass"
+
+    num_state_variables = 2
+    num_noise_variables = 2
+    coupling_variables = {0: "x", 1: "y"}
+    state_variable_names = ["x", "y"]
+    required_parameters = ["a", "omega", "ext_input_x", "ext_input_y"]
+    required_couplings = ["network_x", "network_y"]
+
+    def __init__(self, parameters=None):
+        super().__init__(parameters=parameters or DEFAULT_PARAMS)
+
+    def _initialize_state_vector(self):
+        """
+        Initialize state vector.
+        """
+        self.initial_state = [0.5 * np.random.uniform(-1, 1)] * self.num_state_variables
+
+    def _derivatives(self, coupling_variables):
+        [x, y] = self._unwrap_state_vector()
+
+        d_x = (
+            (self.parameters["a"] - x ** 2 - y ** 2) * x
+            - self.parameters["omega"] * y
+            + coupling_variables["network_x"]
+            + system_input(self.noise_input_idx[0])
```

Comment:
`system_input` is the way how to insert input - noise and stimulus.. by default this is used for noise, but in the thalamic model (soon to come in PR2) I used it for a sinusoidal and square drive (see `model_input.py`)

P.S system input can contain delays (!!) but they are not supported as of now.. just might be nice future addition